semgrep-core/Matching/Generic_vs_generic.ml

```diff
@@ -910,6 +910,28 @@ and m_literal a b =
     -> fail ()
 (*e: function [[Generic_vs_generic.m_literal]] *)
 
+and m_wrap_m_int_opt (a1, a2) (b1, b2) =
+  match a1, b1 with
+  (* iso: semantic equivalence of value! 0x8 can match 8 *)
+  | Some i1, Some i2 when i1 =|= i2 -> return ()
+  (* if the integers (or floats) were too large or were using
+   * a syntax OCaml int_of_string could not parse,
+   * we default to a string comparison *)
+  | _ ->
+      let a1 = Parse_info.str_of_info a2 in
+      let b1 = Parse_info.str_of_info b2 in
+      m_wrap m_string (a1, a2) (b1, b2)
+
+and m_wrap_m_float_opt (a1, a2) (b1, b2) =
+  match a1, b1 with
+  (* iso: semantic equivalence of value! 0x8 can match 8 *)
```

Comment:
Minor thing, the example for floats could be `0.123e1 can match 1.23` for example.